File: ToolBag/ParseTool/src/parsetool/dataview/models/condition/Compare_Complex.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

package parsetool.dataview.models.condition;

import parsetool.dataview.models.condition.exists.Existed_Compare_Prior;
import parsetool.dataview.models.condition.in.In_Expression_Prior;
import parsetool.models.common.TokenPair;

/**
 *
 * @author Administrator
 */
public class Compare_Complex extends TokenPair{
    private Binary_Prior bianary_prior;
    private In_Expression_Prior in_expression_prior;
    private Between_Prior between_prior;
    private Existed_Compare_Prior existed_compare_prior;

    /**
     * @return the bianary_prior
     */
    public Binary_Prior getBianary_prior() {
        return bianary_prior;
    }

    /**
     * @param bianary_prior the bianary_prior to set
     */
    public void setBianary_prior(Binary_Prior bianary_prior) {
        this.bianary_prior = bianary_prior;
    }

    /**
     * @return the in_expression_prior
     */
    public In_Expression_Prior getIn_expression_prior() {
        return in_expression_prior;
    }

    /**
     * @param in_expression_prior the in_expression_prior to set
     */
    public void setIn_expression_prior(In_Expression_Prior in_expression_prior) {
        this.in_expression_prior = in_expression_prior;
    }

    /**
     * @return the between_prior
     */
    public Between_Prior getBetween_prior() {
        return between_prior;
    }

    /**
     * @param between_prior the between_prior to set
     */
    public void setBetween_prior(Between_Prior between_prior) {
        this.between_prior = between_prior;
    }

    /**
     * @return the existed_compare_prior
     */
    public Existed_Compare_Prior getExisted_compare_prior() {
        return existed_compare_prior;
    }

    /**
     * @param existed_compare_prior the existed_compare_prior to set
     */
    public void setExisted_compare_prior(Existed_Compare_Prior existed_compare_prior) {
        this.existed_compare_prior = existed_compare_prior;
    }
    
}

File: CodeHelper/CodeHelper.Core/Parse/ParseResults/DataViews/Condition/Compare_Complex.cs
using System;
using CodeHelper.Core.Parse.ParseResults.DataViews.Condition.In;
using CodeHelper.Core.Parse.ParseResults.DataViews.Condition.Exists;
namespace CodeHelper.Core.Parse.ParseResults.DataViews.Condition
{
    public class Compare_Complex : TokenPair
    {
        public Binary_Prior Bianary_prior { get; set; }
        public In_Expression_Prior In_expression_prior { get; set; }
        public Between_Prior Between_prior { get; set; }
        public Existed_Compare_Prior Existed_compare_prior { get; set; }

        public ConditionInfo Parse(SelectAtomContext ctx)
        {
            if (Bianary_prior != null)
                return Bianary_prior.Parse(ctx);

            if (In_expression_prior != null)
                return In_expression_prior.Parse(ctx);

            if (Between_prior != null)
                return Between_prior.Parse(ctx);

            if (Existed_compare_prior != null)
                return Existed_compare_prior.Parse(ctx);

            return null;
        }
    }
}